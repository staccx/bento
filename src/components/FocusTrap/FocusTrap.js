import React from "react"
import themeProps from "./FocusTrap.themeProps"
import styled from "styled-components"
import { applyVariants } from "../../theming"
import { useFocusTrap } from "../../hooks/useFocusTrap/useFocusTrap"

/**
 * FocusTrap that lets you keep focus within it.
 */
const FocusTrap = React.forwardRef(
  ({ children, options = {}, activateOnReady = true }, ref) => {
    const refToTrapAround = React.useRef(null)
    useFocusTrap(
      refToTrapAround,
      {
        initialFocusElement: ref,
        activateOnReady
      },
      options
    )

    return (
      <FocusTrapContainer ref={refToTrapAround}>{children}</FocusTrapContainer>
    )
  }
)

const FocusTrapContainer = styled.div`
  ${applyVariants(themeProps.container)};
`

FocusTrap.themeProps = themeProps

export default FocusTrap
