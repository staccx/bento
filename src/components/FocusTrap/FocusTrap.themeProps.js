import { themePropTypes } from "../../theming"

const themeProps = {
  container: {
    name: "FOCUS_TRAP",
    description: "Focus trap container",
    type: themePropTypes.component
  }
}

export default themeProps
