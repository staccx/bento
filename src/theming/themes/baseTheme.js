import React from "react"
import Theme from "../Theme"
import reset from "../styles/cssResets"
import grids from "../../components/Layout/Layout/Layout.Grids"
import codeTheme from "../codeTheme"

const Logo = () => <div>ThemeComponent for BaseTheme</div>
Logo.displayName = "Logo for base theme"

export const theme = {
  name: "Base Theme",
  storybookName: "Base Theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190", // for toggle eks
    link: "#279AF1"
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px",
    lineHeight: "1.5"
  },
  fontWeight: {
    bold: "bold",
    normal: "normal",
    light: "lighter"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px",
  reset,
  grid: grids,
  logo: Logo
}

export const StyleguideTheme = new Theme(theme, codeTheme)

export default theme
