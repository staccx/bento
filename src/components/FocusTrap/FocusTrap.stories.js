import React from "react"
import FocusTrap from "./FocusTrap"

export default {
  title: "New/FocusTrap",
  component: FocusTrap
}

export const FocusTrapDefault = args => (
  <div tabIndex="0">
    This is outside the focus trap
    <FocusTrap {...args}>
      <>
        <span>This is within</span>
        <div>
          <p>Some more text</p>
        </div>
        <button>This is focusable</button>
        <button>This is focusable</button>
      </>
    </FocusTrap>
  </div>
)
FocusTrapDefault.args = { options: { clickOutsideDeactivates: false } }
