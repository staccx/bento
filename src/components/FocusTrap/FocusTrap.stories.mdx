import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import FocusTrap from "./FocusTrap.js"

<Meta title="components/FocusTrap" component={FocusTrap} parameters={{component: FocusTrap.themeProps.container.name}} />

# FocusTrap

FocusTrap that lets you keep focus within it.


```javascript
import {FocusTrap} from "@staccx/bento"
```

## FocusTrap default
<Preview>
  <Story name="FocusTrap default">
  <div tabindex="0">
  This is outside the focus trap
    <FocusTrap options={{clickOutsideDeactivates:false}}>
    <>
        <span>This is within</span>
            <div>
                <p>Some more text</p>
            </div>
            <button>
                This is focusable
                </button>
                 <button>
                                This is focusable
                                </button>
                </>
    </FocusTrap>
    </div>
  </Story>
</Preview>

