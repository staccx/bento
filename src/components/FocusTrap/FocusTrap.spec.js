import React from "react"
import FocusTrap from "./FocusTrap"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../theming/themes/baseTheme"

describe("FocusTrap", () => {
  describe("Snapshots", () => {
    it("First", () => {
      const three = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <div>
              <p>this is outside of the trap</p>
              <FocusTrap options={{ clickOutsideDeactivates: false }}>
                <>
                  <span>this is inside the trap</span>
                  <div>
                    <p>Some more text</p>
                  </div>
                  <button>This is focusable</button>
                  <button>This is focusable</button>
                </>
              </FocusTrap>
            </div>
          </ThemeProvider>
        )
        .toJSON()
      expect(three).toMatchSnapshot()
    })
  })
})
