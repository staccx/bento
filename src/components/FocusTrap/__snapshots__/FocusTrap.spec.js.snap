// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FocusTrap Snapshots First 1`] = `
<div>
  <p>
    this is outside of the trap
  </p>
  <div
    className="sc-AxjAm jzbbdJ"
  >
    <span>
      this is inside the trap
    </span>
    <div>
      <p>
        Some more text
      </p>
    </div>
    <button>
      This is focusable
    </button>
    <button>
      This is focusable
    </button>
  </div>
</div>
`;
